### ISSUE TITLE
$auth.setToken('12341234.....') throws exception

### ISSUE THREAD

**Issue by ribomation** [sahat/satellizer#302]
`$auth.getToken()` returns a token-string, which is expected. However, `$auth.setToken(tokenString)` expects something different than a string. I get the exception `Cannot read property 'false' of undefined`. 

It looks to me that the expression (in `shared.setToken`) below on line 277-278, always crashes.
```
var token = response.access_token || config.tokenRoot && response.data[config.tokenRoot] ?
    response.data[config.tokenRoot][config.tokenName] : response.data[config.tokenName];
```

The public function wraps the token-string with key `access_token`
```
$auth.setToken = function(token, redirect) {
    shared.setToken({ access_token: token }, redirect);
};
```
which triggers the then-part of the ?: expression. `config.tokenRoot` is set to false by default hence the crash. Because the payload sent to `shared.setToken` is {access_token: ...} it can never succeed getting `response.data[...]`.

Or, something very subtle in JS semantics is expected and passes me by. 

Anyway, for consistency I do recommend that getToken() returns a value that can be passed in to setToken().

**Comment by vitalyrotari** [comment]
@ribomation Have same problem

**Comment by mcantrell** [comment]
+1 for the pull request from @vitalyrotari (fixed the issue for me)

**Comment by sahat** [comment]
Fixed in #306.

**Comment by jbeckton** [comment]
The code fix implemented for #306 seems to cause problems as well. I have a .Net Web API REST back end and the http response never has the payload from the server in the root of the object. You always have to look at response.data. I figured this is an http thing and not particular to any back end technology.

In my case Shared.setToken can never work because the only time it actually looks at response.data is when there is a value for config.tokenRoot.

if I change var accessToken = response && response.access_token;
to var accessToken = response && response.data.access_token;

it works fine
